extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/data/GreenbidsInferenceDataService.java

```diff
@@ -0,0 +1,194 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.data;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CountryResponse;
+import com.maxmind.geoip2.record.Country;
+import lombok.Value;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Value(staticConstructor = "of")
```

Comment:
Removed `@Values` annotation + checked other classes. 

In `GreenbidsInvocationResult` I prefer to keep annotation because its fields (updatedBidRequest, invocationAction, analyticsResult) depend on each call so there is no option to inject is as bean in spring config. So we discussed it with Anton to be able to keep the static constructor with `@Values` and define invocation results